downloader/src/main/kotlin/Downloader.kt

```diff
@@ -359,28 +359,36 @@ class Downloader(private val config: DownloaderConfiguration) {
             }
         }
 
-        try {
-            if (sourceArchive.extension == "gem") {
-                // Unpack the nested data archive for Ruby Gems.
-                val gemDirectory = createOrtTempDir("gem")
-                val dataFile = gemDirectory / "data.tar.gz"
-
-                try {
-                    sourceArchive.unpack(gemDirectory)
-                    dataFile.unpack(outputDirectory)
-                } finally {
-                    gemDirectory.safeDeleteRecursively()
+        if (sourceArchive.extension == "gem") {
+            // Unpack the nested data archive for Ruby Gems.
+            val gemDirectory = createOrtTempDir("gem")
+            val dataFile = gemDirectory / "data.tar.gz"
+
+            try {
+                sourceArchive.unpack(gemDirectory)
+                dataFile.unpack(outputDirectory)
+            } catch (e: IOException) {
+                logger.error {
+                    "Could not unpack source artifact '${sourceArchive.absolutePath}': ${e.collectMessages()}"
                 }
-            } else {
-                sourceArchive.unpackTryAllTypes(outputDirectory)
-            }
-        } catch (e: IOException) {
-            logger.error {
-                "Could not unpack source artifact '${sourceArchive.absolutePath}': ${e.collectMessages()}"
+
+                tempDir?.safeDeleteRecursively()
+                throw DownloadException(e)
+            } finally {
+                gemDirectory.safeDeleteRecursively()
             }
+        } else {
+            try {
+                sourceArchive.unpackTryAllTypes(outputDirectory)
+            } catch (e: IOException) {
+                logger.warn {
+                    "Could not unpack source artifact '${sourceArchive.absolutePath}': ${e.collectMessages()}"
+                }
 
-            tempDir?.safeDeleteRecursively()
-            throw DownloadException(e)
+                logger.info {
```

Comment:
Should we maybe only do this if an Apache Tika check says it's a text file? On the other hand, there might be value in scanning e.g. single TTF files, too.

Reply:
> Would this edge case be solved with the tika validation?

It depends on the Mime types that we choose to accept, I guess.

Reply:
We could accept Mime-types that `.startsWith("text/")` but exclude `"text/html"` and include `"application/javascript"`. What do you think @MarcelBochtler?